### PR TITLE
compat: add window_properties object to json output of views

### PR DESCRIPTION
**Pull request by Snaipe**
Some existing tools that rely on i3's json output, like the i3ipc
python module and scripts using it, break when they can't find the
window_properties object in a view.

To achieve a better drop-in replacement compatibility with i3 and
existing tools for it, we add the window_properties object to
the output of get_tree, along with the "class", "instance", and "title"
sub-fields. These three values are set to the view's class, app_id,
and title respectively.

In addition, if the view does not have a class, window_properties.class
is set to the app_id, to avoid breaking tools that expect all windows
to have a class in the X world.

Signed-off-by: Franklin "Snaipe" Mathieu <me@snai.pe>

### CODE REVIEW

File: sway/ipc-json.c [hunk 1]
```diff
@@ -268,6 +268,28 @@ static void ipc_json_describe_view(struct sway_container *c, json_object *object
...
+	 * The "window_properties" object is mostly redundant with other existing
+	 * fields in the view object, but is still somewhat necessary for drop-in
+	 * compatibility with existing i3 ipc tools/libs that expect them
+	 * (e.g. the i3ipc python module).
+	 */
+	json_object *window_props = json_object_new_object();
+
+	// Some tools for i3 break when windows don't have a class.
+	if (!class) {
+		class = app_id;
```
[the remark this answers is not in the document]
Comment (RyanDwyer) [reply sway/ipc-json.c]: -1 to this. Loading the app id into the class or instance fields makes it confusing to end users. I value transparency - that is: telling the user what the properties actually are - over trying to allow for a zero configuration port between i3 and sway. I add an `app_id` property, set it accordingly, and let the class and instance be null. This is the approach we've taken with criteria too.

[the remark this answers is not in the document]
Comment (Snaipe) [reply sway/ipc-json.c]: Wow, sorry for the spam, looks like [GitHub is choking on requests right now](https://blog.github.com/2018-10-21-october21-incident-report/).

[the remark this answers is not in the document]
Comment (RyanDwyer) [reply sway/ipc-json.c]: I don't have a problem with setting I3SOCK because there's no bad consequences for doing it. But putting the app id into the class and instance is a decision that we could be forced to live with for a while, can confuse and frustrate users, and will become a legacy (especially when users no longer run Xwayland).

[the remark this answers is not in the document]
Comment (progandy) [reply sway/ipc-json.c]: As a compromise, maybe set the class to _wayland_ if it is a native wayland window?

[the remark this answers is not in the document]
Comment (Snaipe) [reply sway/ipc-json.c]: Alright. I somewhat find it just as frustrating to see every single ipc script that check the instance or the class of a window completely break, but I can see why you wouldn't want that in sway. I just hoped that sway would own being its own thing for the sake of consistency.

I'll just write a standalone ipc proxy providing an I3SOCK socket to convert from the sway-flavored json to the i3-flavored one in the meantime.

tl;dr: Dropping this PR; window_properties are very much more so a compatibility thing with i3 rather than something critical.
